### PR TITLE
orb-ui: simulations & show-car

### PR DESCRIPTION
**Pull request by fouge**
easy ways to test

```
orb-ui simulation <mode>
```

simulations modes:
- `operator` legacy flow
- `self-serve` 
- `show-car`
   - to test: `sudo echo 441 > /sys/class/gpio/export` to set button as GPIO input

### CODE REVIEW

File: orb-ui/src/dbus.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ impl Interface {
     /// Forward events to UI engine by sending serialized engine::Event to the event channel.
     async fn orb_signup_state_event(&mut self, event: String) -> zbus::fdo::Result<()> {
         // parse event to engine::Event using json_serde
-        tracing::debug!("received JSON event: {}", event);
+        tracing::trace!("received JSON event: {}", event);
```
Comment (andronat) [inline orb-ui/src/dbus.rs:25]: Do we have tracing enabled? I think given the big changes we do atm it's important to get max info.

**Review by andronat — APPROVED** [review]

File: orb-ui/src/dbus.rs (hunk 1 above)
Comment (TheButlah) [reply orb-ui/src/dbus.rs:25]: The solution is probably to change RUST_LOG in the .service file for now. The code should be written with the expectation that anything below INFO is omitted, DEBUG is not high frequency, and TRACE is full verbosity of all everything.

File: orb-ui/src/main.rs [hunk 2]
```diff
@@ -49,23 +49,45 @@ enum SubCommand {
...
+
+    /// show-car, infinite loop of signup
+    #[clap(action)]
+    ShowCar,
+}
+
 static HW_VERSION_FILE: OnceLock<String> = OnceLock::new();
 
-fn get_hw_version() -> Result<String> {
+#[derive(Debug, PartialEq)]
```
Comment (TheButlah) [inline orb-ui/src/main.rs:77]:
```suggestion
#[derive(Debug, PartialEq, Eq, Clone, Copy)]
```

File: orb-ui/src/simulation.rs [hunk 3]
```diff
@@ -1,90 +1,229 @@
...
+                biometric_capture_error = true;
+                ui.signup_fail(SignupFailReason::Timeout);
+                break;
+            }
+
+            // simulate gimbal movement
+            let x_base = if i < 50 { 40000 } else { 35000 };
+            let x_rand = rand::random::<i32>() % 1000;
+            let y_rand = rand::random::<i32>() % 1000;
+            let x_sign: i32 = if rand::random::<u8>() % 2 == 0 { 1 } else { -1 };
```
Comment (TheButlah) [inline orb-ui/src/simulation.rs:151]: Probably more straightforward to do `rand::random::<bool>()`

Also, x_rand is already a random signed number. Whats the point of `x_sign` then?

File: orb-ui/src/simulation.rs [hunk 4]
```diff
@@ -1,90 +1,229 @@
...
+                    time::sleep(Duration::from_millis(500)).await;
+                }
+                ui.biometric_pipeline_success();
+
+                time::sleep(Duration::from_secs(1)).await;
+                if rand::random::<u8>() % 2 == 0 {
+                    ui.signup_success();
+                } else {
+                    let fail_reason = SignupFailReason::from(
+                        rand::random::<u8>() % SignupFailReason::Unknown as u8,
```
Comment (TheButlah) [inline orb-ui/src/simulation.rs:192]: Maybe consider doing this: `https://www.reddit.com/r/rust/comments/i5n2zm/comment/g0rb9ob/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button`

The way its done right now feels a bit hacky, but if you're opinionated, we can keep it.

File: orb-ui/src/simulation.rs [hunk 5]
```diff
@@ -1,90 +1,229 @@
...
-        let fail_reason = SignupFailReason::from(
-            rand::random::<u8>() % SignupFailReason::Unknown as u8,
-        );
-        ui.signup_fail(fail_reason);
+        info!("Waiting for button press");
+        if showcar {
+            // check that file exists
+            if !std::path::Path::new("/sys/class/gpio/PQ.06/value").exists() {
+                // write 441 into /sys/class/gpio/export
+                if let Err(e) = std::fs::write("/sys/class/gpio/export", "441") {
```
Comment (TheButlah) [inline orb-ui/src/simulation.rs:79]: blocking in async. Use tokio.

File: orb-ui/src/simulation.rs [hunk 6]
```diff
@@ -1,90 +1,229 @@
...
+                    error!("Button cannot be configured, use `sudo echo 441 > /sys/class/gpio/export`: {}", e);
+                }
+                break;
+            }
+
+            loop {
+                // read cat /sys/class/gpio/PQ.06/value in a loop until it's 1
+                if let Ok(value) =
+                    std::fs::read_to_string("/sys/class/gpio/PQ.06/value")
```
Comment (TheButlah) [inline orb-ui/src/simulation.rs:88]: blocking in async.

File: orb-ui/src/main.rs [hunk 7]
```diff
@@ -49,23 +49,45 @@ enum SubCommand {
...
     let hw_file = HW_VERSION_FILE.get_or_init(|| {
         env::var("HW_VERSION_FILE")
             .unwrap_or_else(|_| "/usr/persistent/hardware_version".to_string())
     });
-    debug!("Reading HW version from {}", hw_file.as_str());
+    debug!("Reading hardware version from {}", hw_file.as_str());
 
-    String::from_utf8(
+    let hw =String::from_utf8(
         fs::read(hw_file.as_str())
```
Comment (TheButlah) [inline orb-ui/src/main.rs:91]: blocking in async. Although this happens at the start of the program and will eventually unblock, its still a potential footgun/hard to debug thing that may bite us later, so it makes sense to fix it.

**Review by TheButlah — CHANGES_REQUESTED** [review]

File: orb-ui/src/simulation.rs (hunk 3 above)
Comment (fouge) [reply orb-ui/src/simulation.rs:151]: I thought the modulo wouldn't work for negative number, it does :)

File: orb-ui/src/simulation.rs (hunk 4 above)
Comment (fouge) [reply orb-ui/src/simulation.rs:192]: it's quite straightforward for me but i've been writing C my whole life 😄

Comment (fouge) [reply orb-ui/src/simulation.rs:192]: done

File: orb-ui/src/dbus.rs (hunk 1 above)
Comment (fouge) [reply orb-ui/src/dbus.rs:25]: parsed events are printed as `debug` logs in the Pearl and Diamond engines, this should be kept as `trace`

**Review by TheButlah — APPROVED** [review]

**Pull request merged**